homeassistant/components/hue/sensor.py

```diff
@@ -0,0 +1,324 @@
+
+        for sensor_id, sensor in current.items():
+            if sensor_id not in progress_waiting:
+                sensor.async_schedule_update_ha_state()
+
+        return
+
+    finally:
+        _LOGGER.debug('Finished sensor request in %.3f seconds',
+                      monotonic() - start)
+
+    if not bridge.available:
+        _LOGGER.info('Reconnected to bridge %s', bridge.host)
+        bridge.available = True
+
+    new_sensors = []
+    sensor_device_names = {}
+
+    for item_id in api:
+        if item_id not in current:
+            name = PRESENCE_NAME_FORMAT.format(api[item_id].name)
+            if api[item_id].type == aiohue.sensors.TYPE_ZLL_PRESENCE:
+                s = HuePresence(
+                    hass, api[item_id], name, request_bridge_update,
+                    bridge)
+                sensor_device_names[s.device_id] = api[item_id].name
+                current[item_id] = s
+
+    # Iterate again now we have all the presence sensors, and add the related
+    # sensors with nice names
+    for item_id in api:
+        if item_id not in current:
+            device_id = api[item_id].uniqueid
+            if device_id and len(device_id) > 23:
+                device_id = device_id[:23]
+            name = api[item_id].name
+            if api[item_id].type == aiohue.sensors.TYPE_ZLL_LIGHTLEVEL:
+                if device_id in sensor_device_names:
+                    name = LIGHT_LEVEL_NAME_FORMAT.format(
+                        sensor_device_names[device_id])
+                current[item_id] = HueLightLevel(
+                    hass, api[item_id], name, request_bridge_update, bridge)
+            elif api[item_id].type == aiohue.sensors.TYPE_ZLL_TEMPERATURE:
+                if device_id in sensor_device_names:
+                    name = TEMPERATURE_NAME_FORMAT.format(
+                        sensor_device_names[device_id])
+                current[item_id] = HueTemperature(
+                    hass, api[item_id], name, request_bridge_update, bridge)
+
+            if item_id in current:
+                new_sensors.append(current[item_id])
+
+        elif item_id not in progress_waiting:
+            current[item_id].async_schedule_update_ha_state()
+
+    if new_sensors:
+        async_add_entities(new_sensors)
+
+
+class GenericHueSensor:
+    """Representation of a Hue sensor."""
+
+    def __init__(self, hass, sensor, name, request_bridge_update, bridge):
+        """Initialize the sensor."""
+        self.hass = hass
+        self.sensor = sensor
+        self._name = name
+        self.async_request_bridge_update = request_bridge_update
+        self.bridge = bridge
+
+        if self.swupdatestate == "readytoinstall":
+            err = (
+                "Please check for software updates of the %s "
+                "sensor in the Philips Hue App."
+            )
+            _LOGGER.warning(err, self.name)
+
+        self.entity_id = async_generate_entity_id(
```

Comment:
There is no need for this.

Reply:
even with the custom `_entity_id_format`?

Reply:
There shouldn't be a custom entity ID format. Entity IDs are reserved in the entity registry using the unique ID. Initial one is based on the name, after that the user controls it.